src/transformState.js

```diff
@@ -6,6 +6,30 @@
  */
 function transformState(state, actions) {
   // write code here
+  for (const action of actions) {
+    switch (action['type']) {
+      case 'addProperties':
+        Object.assign(state, action['extraData']);
+        break;
+
+      case 'removeProperties':
+        for (const key of action['keysToRemove']) {
+          delete state[key];
+        }
+        break;
+
+      case 'clear':
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+
+      default:
+        break;
```

Comment:
no sense of keeping an empty `default:` case, but you can make it behave as `clear` i suppose
```suggestion
      default:
        for (const key in state) {
          delete state[key];
        }
```

Reply:
just thought, you could actually keep the default
but don't keep it empty - at least console.log some info or even better `throw new Error('action is not supported')`

Reply:
Sure, thanks